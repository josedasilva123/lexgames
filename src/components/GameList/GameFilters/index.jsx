import React from "react";
import { StyledButton } from "../../../styles/button";
import styles from "./style.module.css";

const GameFilters = ({ categories, filter, setFilter }) => {
  return (
    <div className={styles.categoryList}>
      <StyledButton buttonStyle={(filter === "" ? "solid1" : "outline")} onClick={() => setFilter("")}>Todos</StyledButton>
     
      {categories.map((category) => (
        <StyledButton key={category} buttonStyle={(filter === category ? "solid1" : "outline")} onClick={() => setFilter(category)}>
          {category}
        </StyledButton>
      ))}
    </div>
  );
};

export default GameFilters;
