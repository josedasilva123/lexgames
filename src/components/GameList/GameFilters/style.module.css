.categoryList{
    display: flex;
    flex-wrap: wrap;
    gap: 10px;
    margin-bottom: 20px;
}

.categoryList li{
    white-space: nowrap;
}

@media (max-width: 800px){
    .categoryList{
        flex-wrap: unset;
        overflow-x: scroll;
        padding-bottom: 8px;
        margin-right: -16px;
    }
}