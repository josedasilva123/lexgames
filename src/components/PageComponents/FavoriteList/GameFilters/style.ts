import styled from "styled-components";

export const StyledGameFilters = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 10px;
  margin-bottom: 20px;

  li {
    white-space: nowrap;
  }

  @media (max-width: 800px) {
    flex-wrap: unset;
    overflow-x: scroll;
    padding-bottom: 8px;
    margin-right: -16px;
  }
`;
