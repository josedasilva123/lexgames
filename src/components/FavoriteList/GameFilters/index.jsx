import React from "react";
import { StyledButton } from "../../../styles/button";
import { StyledGameFilters } from "./style";

const GameFilters = ({ categories, filter, setFilter }) => {
  return (
    <StyledGameFilters>
      <StyledButton buttonStyle={(filter === "" ? "solid1" : "outline")} onClick={() => setFilter("")}>Todos</StyledButton>
     
      {categories.map((category) => (
        <StyledButton key={category} buttonStyle={(filter === category ? "solid1" : "outline")} onClick={() => setFilter(category)}>
          {category}
        </StyledButton>
      ))}
    </StyledGameFilters>
  );
};

export default GameFilters;
