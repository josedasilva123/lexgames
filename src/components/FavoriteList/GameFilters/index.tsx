import React from "react";
import { StyledButton } from "../../../styles/button";
import { StyledGameFilters } from "./style";

interface iGameFiltersProps{
  categories: string[];
  filter: string;
  setFilter: React.Dispatch<React.SetStateAction<string>>;
}

const GameFilters = ({ categories, filter, setFilter }: iGameFiltersProps) => {
  return (
    <StyledGameFilters>
      <StyledButton buttonStyle={(filter === "" ? "solid1" : "outline")} onClick={() => setFilter("")}>Todos</StyledButton>
     
      {categories.map((category) => (
        <StyledButton key={category} buttonStyle={(filter === category ? "solid1" : "outline")} onClick={() => setFilter(category)}>
          {category}
        </StyledButton>
      ))}
    </StyledGameFilters>
  );
};

export default GameFilters;
