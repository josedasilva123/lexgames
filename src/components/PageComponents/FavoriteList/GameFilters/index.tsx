import React, { useContext } from "react";
import { CatalogContext } from "../../../contexts/CatalogContext/CatalogContext";
import { GamesContext } from "../../../contexts/GamesContext/GamesContext";
import { StyledButton } from "../../../styles/button";
import { StyledGameFilters } from "./style";

const GameFilters = () => {
  const { filter, setFilter } = useContext(GamesContext);
  const { newCategories } = useContext(CatalogContext);

  return (
    <StyledGameFilters>
      <StyledButton buttonStyle={(filter === "" ? "solid1" : "outline")} onClick={() => setFilter("")}>Todos</StyledButton>
     
      {newCategories?.map((category) => (
        <StyledButton key={category} buttonStyle={(filter === category ? "solid1" : "outline")} onClick={() => setFilter(category)}>
          {category}
        </StyledButton>
      ))}
    </StyledGameFilters>
  );
};

export default GameFilters;
