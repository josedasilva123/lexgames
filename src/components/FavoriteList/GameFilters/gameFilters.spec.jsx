import { fireEvent, render, screen, waitFor } from "@testing-library/react"
import React from "react"
import GameFilters from "."

const mockCategories = ["RPG", "MMORPG", "Shooter"]

describe("<GameFilters />", () => {
    it("should render with filters options and no selected filter", () => {
        render(
            <GameFilters categories={mockCategories} filter="" />
        )

        const filters = screen.getAllByRole("button");
        const allButton = screen.getByText("Todos");

        expect(filters).toHaveLength(4);
        expect(allButton).toHaveClass("sc-bczRLJ cTyziY");        
    })

    it("should render filters options with Shooter selected", () => {
        render(
            <GameFilters categories={mockCategories} filter="Shooter" />   
        )

        const shooterButton = screen.getByText("Shooter");
        expect(shooterButton).toHaveClass("sc-bczRLJ cTyziY");    
    })

    it("should fire a setState action when button event is fired", async () => {
        const setFilterMock = jest.fn();

        render(
            <GameFilters categories={mockCategories} filter="Shooter" setFilter={setFilterMock} />   
        )      

        const allButton = screen.getByText("Todos");
        fireEvent.click(allButton);
  

        await waitFor(() => {
            expect(setFilterMock).toHaveBeenCalledTimes(1);
        })
    })

    it("should match with snapshot", () => {
        const { container } = render(
            <GameFilters categories={mockCategories} filter="" />
        )

        expect(container).toMatchSnapshot();
    })
})