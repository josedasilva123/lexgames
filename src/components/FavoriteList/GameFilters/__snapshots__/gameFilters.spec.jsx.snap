// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameFilters /> should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi fmfXIE"
  >
    <button
      class="sc-bczRLJ cTyziY"
    >
      Todos
    </button>
    <button
      class="sc-bczRLJ dXjvqJ"
    >
      RPG
    </button>
    <button
      class="sc-bczRLJ dXjvqJ"
    >
      MMORPG
    </button>
    <button
      class="sc-bczRLJ dXjvqJ"
    >
      Shooter
    </button>
  </div>
</div>
`;
